### ISSUE TITLE
TODO missing when in nested code block.

### ISSUE THREAD

**Issue by Fran-B** [doxygen/doxygen#6763]
I discovered a number of missing "todo" items and traced the issue to the fact that they are placed within conditional code blocks.
    doxygen -- version = 1.6.1
    OS = Red Hat Enterprise Linux Workstation release 6.10 (Santiago)

basic example is
`
void foo()
{
  ///@todo is visible
  if (1 == 1)
  {
      ///@todo is not visible
  }
}
`

Certainly a "todo" needs to be placed directly where it has meaning so it needs to be able to go most anywhere.
Attaching Doxyfile and full class example.

[todoExample.tar.gz](https://github.com/doxygen/doxygen/files/2746076/todoExample.tar.gz)

**Comment by albert-github** [comment]
You are using doxygen 1.6.1 (i.e. August 2009)? What happens with the current version: 1.8.15 ?

**Comment by Fran-B** [comment]
Sorry, I didn't realize that we were so far behind in versions.
I do not much control over what version I am using.  It will take me a while to figure out how to get that to work.
Thinking that it would probably be easier for somebody who is already using the current version, I just realized that the two source files will need to be moved into an "src" directory for my Doxyfile to work.

Meanwhile, I will check with a sys-admin to see what I need to do to get the latest version available for me to try.

**Comment by albert-github** [comment]
Red Hat normally is lagging a bit with the doxygen version. Easiest is to download the release version from http://doxygen.nl/download.html or clone the doxygen repository and in both case build doxygen.

**Comment by albert-github** [comment]
In your example I see on the `todo page` 16 `todo` entries, the number I would expect (dn the code I see 19 `@todo` statements, but 3 are hidden by `#if` constructs).

Can you please indicate which ones you are missing?

**Comment by Fran-B** [comment]
Sorry that I missed the last comment by Albert.  I did not expect to see the "todo" entries hidden by #ifdef, only those that would be visible to the compiler.  That is what I saw when I retested with the newer release.

One would normally expect comments in code eliminated from compilation via #if construct to not be visible to Doxygen either.  Thus if the source were being used by two projects, project-specific customizations done in that manner would only be visible in the ToDo list for the project that the code is compiled into.

Like the later version works exactly as I would have expected.

Sorry for the trouble.
